diff -uNr kernel-3.10.0-514.6.1.el7/linux-3.10.0-514.6.1.el7.x86_64/drivers/pci/xen-pcifront.c kernel-3.10.0-514.6.1.el7.pcifront/linux-3.10.0-514.6.1.el7.x86_64/drivers/pci/xen-pcifront.c
--- kernel-3.10.0-514.6.1.el7/linux-3.10.0-514.6.1.el7.x86_64/drivers/pci/xen-pcifront.c	2016-12-10 10:27:13.000000000 -0500
+++ kernel-3.10.0-514.6.1.el7.pcifront/linux-3.10.0-514.6.1.el7.x86_64/drivers/pci/xen-pcifront.c	2017-01-20 20:04:15.941923816 -0500
@@ -446,6 +446,7 @@
 				 unsigned int domain, unsigned int bus)
 {
 	struct pci_bus *b;
+	LIST_HEAD(resources);
 	struct pcifront_sd *sd = NULL;
 	struct pci_bus_entry *bus_entry = NULL;
 	int err = 0;
@@ -470,14 +471,17 @@
 		err = -ENOMEM;
 		goto err_out;
 	}
+	pci_add_resource(&resources, &ioport_resource);
+	pci_add_resource(&resources, &iomem_resource);
 	pcifront_init_sd(sd, domain, bus, pdev);
 
-	b = pci_scan_bus_parented(&pdev->xdev->dev, bus,
-				  &pcifront_bus_ops, sd);
+	b = pci_scan_root_bus(&pdev->xdev->dev, bus,
+				&pcifront_bus_ops, sd, &resources);
 	if (!b) {
 		dev_err(&pdev->xdev->dev,
 			"Error creating PCI Frontend Bus!\n");
 		err = -ENOMEM;
+		pci_free_resource_list(&resources);
 		goto err_out;
 	}
 
@@ -485,7 +489,7 @@
 
 	list_add(&bus_entry->list, &pdev->root_buses);
 
-	/* pci_scan_bus_parented skips devices which do not have a have
+	/* pci_scan_root_bus skips devices which do not have a have
 	* devfn==0. The pcifront_scan_bus enumerates all devfn. */
 	err = pcifront_scan_bus(pdev, domain, bus, b);
 
